src/platform/Darwin/DnssdImpl.h

```diff
@@ -222,17 +225,50 @@ struct InterfaceInfo
     DnssdService service;
     std::vector<Inet::IPAddress> addresses;
     std::string fullyQualifiedDomainName;
+    bool isDNSLookUpRequested = false;
+};
+
+struct InterfaceKey
+{
+    InterfaceKey()  = default;
+    ~InterfaceKey() = default;
+    inline bool operator<(const InterfaceKey & other) const
+    {
+        return (this->interfaceId < other.interfaceId) ||
+            ((this->interfaceId == other.interfaceId) && (this->hostname < other.hostname));
+    }
+
+    uint32_t interfaceId;
+    std::string hostname;
+    bool isSRPTypeRequested = false;
+};
+
+struct ResolveContextWithType
+{
+    ResolveContextWithType()  = default;
+    ~ResolveContextWithType() = default;
+
+    ResolveContext * context;
+    bool isSRPType = false;
 };
 
 struct ResolveContext : public GenericContext
 {
     DnssdResolveCallback callback;
-    std::map<uint32_t, InterfaceInfo> interfaces;
+    std::map<InterfaceKey, InterfaceInfo> interfaces;
     DNSServiceProtocol protocol;
     std::string instanceName;
     std::shared_ptr<uint32_t> consumerCounter;
     BrowseContext * const browseThatCausedResolve; // Can be null
 
+    // Indicates whether the timer for 250 msecs should be started
+    // to give the resolve on SRP domain some extra time to complete.
+    bool shoulStartSRPTimerForResolve = false;
+    bool isSRPTimerRunning            = false;
+
+    ResolveContextWithType resolveContextWithSRPType;
+    ResolveContextWithType resolveContextWithNonSRPType;
```

Comment:
Note that you could initialize them here, and then don't need to do it in both constructors....

Reply:
done.